### PR TITLE
Migrates to new version of resource pack

### PR DESCRIPTION
**Pull request by dmccown**
This new version is backed by a REST implementation. We are introducing
resources created for the specific use cases we are trying to solve, as
well as establishing a pattern that authors may follow to add new
resources.

The maintainers list will be updated in the near future with the correct
long term maintainers.

### CODE REVIEW

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    # Insist that you have at least one virtual machine that starts with 'MyVm'
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup').where { name.start_with?('WindowsVm') } do
+      it { should exist }
+    end
+
+## Properties
+
+* `os_disks`, `data_disks`, `vm_names`
+
+### os_disks
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: Should be `os\_disks`

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+Gives a list of OS disk names for all the virtual machines in the resource group.
+
+    its('os_disks.sort') { should eq ['MyDisk1', 'MyDisk2'] }
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: Could we give examples that avoid `.sort`, like:

```
its('os_disks') { should include 'DataDisk1' }
its('os_disks') { should include 'DataDisk2' }
```

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+### data_disks
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: Should be `data\_disks`

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+Gives a list of data disks for all the virtual machines in the resource group.
+
+    its('os_disks.sort') { should eq [['MyDisk1'], ['MyDisk2']] }
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: Could we give examples that avoid `.sort`, like:

```
its('os_disks') { should include 'DataDisk1' }
its('os_disks') { should include 'DataDisk2' }
```

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+Note that it returns an array of arrays. You may flatten them for testing:
+
+    its('os_disks.flatten.sort') { should eq ['MyDisk1', 'MyDisk2'] }
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: Could we give examples that avoid `.sort`, like:

```
its('os_disks') { should include 'DataDisk1' }
its('os_disks') { should include 'DataDisk2' }
```

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+### vm_names
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: Should be `vm\_names`

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+Gives a list of all the virtual machine names in the resource group.
+
+    its('vm_names') { should include('MyVm1', 'MyVm2') }
+
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list
+of available matchers, please visit our [Universal Matchers
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: Newline in link

File: docs/resources/azurerm_virtual_machines.md.erb [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    end
+
+    # Should exist if the filter returns a single virtual machine
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup').where('platform' => 'windows') do
+      it { should exist }
+    end
+
+## Azure Permissions
+
+Your [Service
```
Comment (jerryaldrichiii) [inline docs/resources/azurerm_virtual_machines.md.erb]: New line in link

File: libraries/azure_monitor_log_profile.rb [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    retention_enabled: 'retentionPolicyEnabled',
+  }.freeze
+
+  attr_reader(*ATTRS.keys)
+
+  def initialize(options = { name: 'default' })
+    resp = client.log_profile(options[:name])
+    return if resp.nil? || resp.key?('error')
+
+    @name = resp['name']
```
Comment (jerryaldrichiii) [inline libraries/azure_monitor_log_profile.rb]: Should we align the equals with the below?

File: libraries/azure_monitor_log_profile.rb [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      next if instance_variable_defined?("@#{name}")
+
+      instance_variable_set("@#{name}", fields[api_name])
+    end
+
+    @exists = true
+  end
+
+  def has_log_retention_enabled?
+    !!retention_enabled
```
Comment (jerryaldrichiii) [inline libraries/azure_monitor_log_profile.rb]: Double not?

File: libraries/azure_network_security_group.rb [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+  def destination_port_ranges(properties)
+    return Array(properties['destinationPortRange']) if properties['destinationPortRanges'].empty?
+    return properties['destinationPortRanges'] if properties['destinationPortRange'].empty?
+    properties['destinationPortRanges'].push(properties['destinationPortRange'])
+  end
+
+  def matches_port?(ports, match_port)
+    return true if ports.detect { |p| p =~ /^(#{match_port}|\*)$/ }
+    ports
+      .select { |port| port.include?('-') }
```
Comment (jerryaldrichiii) [inline libraries/azure_network_security_group.rb]: Usually I see the second method on the new line. Not sure if it matters though.

File: libraries/azure_security_center_policy.rb [hunk 12]
```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzureSecurityCenterPolicy < AzurermResource
+  name 'azure_security_center_policy'
+  desc 'Verifies settings for Security Center'
+  example <<-EXAMPLE
```
Comment (jerryaldrichiii) [inline libraries/azure_security_center_policy.rb]: I notice we use both `<<-EXAMPLE` and `"` should we standardize?

File: libraries/azurerm_virtual_machine.rb [hunk 13]
```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+# TODO: we will have a name collision with the existing implementation
```
Comment (jerryaldrichiii) [inline libraries/azurerm_virtual_machine.rb]: Could we reword this to be something we can do? Some action item?

File: libraries/azurerm_virtual_machine_disk.rb [hunk 14]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# TODO: we will have a name collision with the existing implementation
```
Comment (jerryaldrichiii) [inline libraries/azurerm_virtual_machine_disk.rb]: Could we reword this as something we can do?

File: libraries/azurerm_virtual_machine_disk.rb [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    @exists = true
+  end
+
+  def to_s
+    "'#{name}' Disk"
+  end
+
+  def encryption_enabled
+    return nil unless properties
+    !!properties.dig('encryptionSettings', 'enabled')
```
Comment (jerryaldrichiii) [inline libraries/azurerm_virtual_machine_disk.rb]: Double not?

File: libraries/support/azure/management.rb [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+    def get(url:, api_version:)
+      confirm_configured!
+
+      cache.fetch(url) do
+        body = azure_client.get(url, params: { 'api-version' => api_version }).body
+        body.fetch('value', body)
+      end
+    end
+
+    def confirm_configured!
```
Comment (jerryaldrichiii) [inline libraries/support/azure/management.rb]: I thought `!` methods were only labeled as such if they modified the variable passed in or modified some external state.

File: test/functional/verify/controls/azure_network_security_group.rb [hunk 17]
```diff
@@ -0,0 +1,24 @@
+resource_group = attribute('resource_group',         default: nil)
+nsg            = attribute('network_security_group', default: nil)
```
Comment (jerryaldrichiii) [inline test/functional/verify/controls/azure_network_security_group.rb]: Are we aligning the `default:nil`?

File: Rakefile [hunk 18]
```diff
@@ -1,101 +1,173 @@
...
 
-    # Create the plan that can be applied to Azure
-    cmd = format("cd %s/build/ && terraform plan -var 'subscription_id=%s' -var 'client_id=%s' -var 'client_secret=%s' -var 'tenant_id=%s' -var 'storage_account_name=%s' -var 'admin_password=%s' -var 'suffix=%s' -out inspec-azure.plan", integration_dir, creds[:subscription_id], creds[:client_id], creds[:client_secret], creds[:tenant_id], sa_name, admin_password, suffix)
-    sh(cmd)
+    puts stdout
 
-    # Apply the plan on Azure
-    cmd = format("cd %s/build/ && terraform apply inspec-azure.plan", integration_dir)
-    sh(cmd)
+    # TODO: add "warnings" back to list when we switch to using InSpec backend
```
[the remark this answers is not in the document]
Comment (dmccown) [reply Rakefile]: We can remove.

File: docs/resources/azure_resource_groups.md.erb [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+The control will pass if the filter returns at least one result. Use
+`should_not` if you expect zero matches.
+
+    describe azure_resource_groups
+      it { should exist }
+    end
+
+## Azure Permissions
+
+Your [Service
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_resource_groups.md.erb]: It's because the next line is that super long URI, that's how the line break function in vim broke it up. I can probably make it cleaner.

[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_resource_groups.md.erb]: hmm maybe let's old off and see if it becomes an issue with the document builder.

File: docs/resources/azure_security_center_policies.md.erb [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+properties of some or all Azure Security Center Policies.
+
+Security Center Policies are defined for each Resource Group. A Security Center
+Policy called `default` also exists for every subscription.
+
+<br />
+
+## Syntax
+
+An `azure_security_center_policies` resource block uses an optional filter to
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policies.md.erb]: We have example on line 36. Do you think it's worth repeating?

File: docs/resources/azure_security_center_policies.md.erb [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+## Examples
+
+### Check for a Security Center Policy
+
+    describe azure_security_center_policies do
+      its('names') { should incldue 'default' }
+    end
+
+### Insist default Security Center Policy exists
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policies.md.erb]: Assert instead of Insist?

File: docs/resources/azure_security_center_policy.md.erb [hunk 22]
```diff
@@ -0,0 +1,206 @@
...
+
+<br />
+
+## Syntax
+
+An `azure_security_center_policy` resource block identifies a Security Center
+Policy by name. In the current Rest API you may only lookup a `default` policy.
+If no policy is given the default one will be used.
+
+    describe azure_security_center_policy(name: 'default') do
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policy.md.erb]: `'name' => 'default'` is used in `.where` as the table data is keyed based on strings. In this case we are calling a constructor that can use a hash with symbols.

File: docs/resources/azure_security_center_policy.md.erb [hunk 23]
```diff
@@ -0,0 +1,206 @@
...
+### Test the default Security Center Policy exists
+
+    describe azure_security_center_policy(name: 'default') do
+      it { should exist }
+    end
+
+### Test the default Security Center Policy has log collection enabled
+
+    describe azure_security_center_policy(name: 'default') do
+      its('log_collection') { should eq('On') }
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policy.md.erb]: Yup, in this case `On` is capitalized.

File: docs/resources/azure_security_center_policy.md.erb [hunk 24]
```diff
@@ -0,0 +1,206 @@
...
+
+    its('id') { should eq('/subscriptions/<SUBSCRIPTION_ID>/providers/Microsoft.Security/policies/default') }
+
+### name
+
+The name of the Security Center Policy.
+
+    its('name') { should eq('default') }
+
+### log_collection
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policy.md.erb]: I think we only need to escape when there's two underscores. E.g. `my_log_collection` would add emphasis to that string. A single underscore shouldn't change the style as far as I know. Is there another reason we need to escape the underscore?

File: docs/resources/azure_security_center_policy.md.erb [hunk 25]
```diff
@@ -0,0 +1,206 @@
...
+
+Log collection indicates if the monitoring agent will collect security data (`On`|`Off`).
+
+    its('log_collection') { should eq('On') }
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policy.md.erb]: We could add a new matcher if that would be more consistent. I'd recommend we do that as a followup.

File: docs/resources/azure_security_center_policies.md.erb (hunk 20 above)
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azure_security_center_policies.md.erb]: Not sure, I just notice we mentioned a filter then don't give an example. Up to you :smile:

File: docs/resources/azure_security_center_policy.md.erb [hunk 26]
```diff
@@ -0,0 +1,206 @@
...
+### contact_phone
+
+Contact phone contains the security contact phone number.
+
+    its('contact_phone') { should eq('1-111-111-1111') }
+
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list
+of available matchers, please visit our [Universal Matchers
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policy.md.erb]: Fixed to be safe.

File: docs/resources/azure_security_center_policies.md.erb (hunk 21 above)
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azure_security_center_policies.md.erb]: Yeah, I use `Assert that` but `Insist that` would be fine here I think.

File: docs/resources/azure_security_center_policy.md.erb (hunk 22 above)
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azure_security_center_policy.md.erb]: Ah, sounds good.

File: docs/resources/azure_security_center_policy.md.erb (hunk 24 above)
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azure_security_center_policy.md.erb]: I've done it with the other docs even if there is just one underscore. I'm not sure to be honest, but we escape all underscores in the core docs.

File: docs/resources/azure_security_center_policy.md.erb (hunk 25 above)
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azure_security_center_policy.md.erb]: Yup, same. Won't block this PR over it. :smile:

File: docs/resources/azurerm_virtual_machine.md.erb [hunk 27]
```diff
@@ -0,0 +1,173 @@
...
+    end
+
+<br />
+
+## Examples
+
+### Ensure that the virtual machine has the expected data disks
+
+    describe azurerm_virtual_machine(resource_group: 'MyResourceGroup', name: 'MyVmName') do
+      its('data_disks.sort') { should eq ['DataDisk1', 'DataDisk2'] }
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azurerm_virtual_machine.md.erb]: You could, but that doesn't assert the same thing. The example I have would assert that you only have those two entries returned. 

You example would still be passing if there are other disks returned. We'd need to add another assertion on the number of disks returned.

[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azurerm_virtual_machine.md.erb]: This would avoid having to use `.sort`

File: docs/resources/azure_security_center_policy.md.erb (hunk 24 above)
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policy.md.erb]: Discussed and decided it's best to escape so we can be consistent.

File: docs/resources/azurerm_virtual_machine.md.erb [hunk 28]
```diff
@@ -0,0 +1,173 @@
...
+    end
+
+### Ensure that the virtual machine has the expected monitoring agent installed
+
+    describe azurerm_virtual_machine(resource_group: 'MyResourceGroup', name: 'MyVmName') do
+      its('monitoring_agent_installed') { should eq true }
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azurerm_virtual_machine.md.erb]: I actually already have a matcher, so I can do:

```
it { should have_monitoring_agent_installed }
```
I'll switch the example.

File: docs/resources/azurerm_virtual_machine_disk.md.erb [hunk 29]
```diff
@@ -0,0 +1,129 @@
...
+
+    /subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/disks/myManagedDisk
+
+### name
+
+The disk's name.
+
+    its('name') { should eq('MyDiskName') }
+
+### managed by
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azurerm_virtual_machine_disk.md.erb]: I think this should be `managedBy` to be accurate with the implementation. The implementation should change so it's `managed_by` though.

File: libraries/azure_monitor_log_profile.rb (hunk 10 above)
Comment (dmccown) [reply libraries/azure_monitor_log_profile.rb]: `!!` converts a value into a boolean. So a falsy value such as `nil` would be false. A truthy value will come back as true.

File: libraries/azure_network_security_group.rb (hunk 11 above)
Comment (dmccown) [reply libraries/azure_network_security_group.rb]: So something like?
```
ports.select { .... }
         .collect { .... }
         .any? { ... }
```

File: libraries/azure_security_center_policy.rb (hunk 12 above)
Comment (dmccown) [reply libraries/azure_security_center_policy.rb]: Generally we've used the HEREDOC syntax (`<<-EXAMPLE`). I think we should standardize on that instead of the multi-line string.

File: libraries/azurerm_virtual_machine_disk.rb (hunk 15 above)
Comment (dmccown) [reply libraries/azurerm_virtual_machine_disk.rb]: See explanation: https://github.com/inspec/inspec-azure/pull/63#discussion_r199280731

File: test/functional/verify/controls/azure_network_security_group.rb (hunk 17 above)
Comment (dmccown) [reply test/functional/verify/controls/azure_network_security_group.rb]: I think so. Is there a spot where it's missing?

File: libraries/azurerm_virtual_machine.rb (hunk 13 above)
Comment (dmccown) [reply libraries/azurerm_virtual_machine.rb]: I'll remove. We resolved it by switching to `azurerm` instead of `azure`. We should probably have a comment on why the name is different or standardize on `azurerm`.

File: libraries/azurerm_virtual_machine_disk.rb (hunk 14 above)
Comment (dmccown) [reply libraries/azurerm_virtual_machine_disk.rb]: https://github.com/inspec/inspec-azure/pull/63#discussion_r199281985

File: test/functional/verify/controls/azure_network_security_group.rb (hunk 17 above)
Comment (dmccown) [reply test/functional/verify/controls/azure_network_security_group.rb]: Ahh I see. Fixed.

File: libraries/azure_security_center_policy.rb (hunk 12 above)
Comment (dmccown) [reply libraries/azure_security_center_policy.rb]: I fixed so we use HEREDOCs everywhere.

File: docs/resources/azurerm_virtual_machine.md.erb (hunk 27 above)
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azurerm_virtual_machine.md.erb]: I'll switch it.

File: docs/resources/azurerm_virtual_machines.md.erb (hunk 5 above)
Comment (dmccown) [reply docs/resources/azurerm_virtual_machines.md.erb]: This one is tricky as we have an array of arrays here. The underlying structure is:
```
[['MyDisk1', 'MyDisk2'], ['MyDisk3']]
```
Do you have a recommendation on how to do this more cleaning in the filter table implementation?

File: libraries/support/azure/management.rb (hunk 16 above)
Comment (dmccown) [reply libraries/support/azure/management.rb]: In this case it's trying to say you may raise an error if things aren't configured correctly.

File: docs/resources/azurerm_virtual_machine_disk.md.erb [hunk 30]
```diff
@@ -0,0 +1,129 @@
...
+
+### sku
+
+The disk's sku name (`Standard_LRS`|`Premium_LRS`).
+
+    its('sku') { should eq('Standard_LRS') }
+
+### properties
+
+The disk's properties. This is a hash of key/value pairs.
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azurerm_virtual_machine_disk.md.erb]: I'll need to noodle on this. We can follow up post merge with an example if that's acceptable.

File: docs/resources/azure_security_center_policies.md.erb (hunk 20 above)
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_security_center_policies.md.erb]: I've generally followed the pattern of the syntax block just shows the describe block without any content. I felt having a syntax example and then an example block got redundant.

File: libraries/support/azure/management.rb (hunk 16 above)
Comment (jerryaldrichiii) [reply libraries/support/azure/management.rb]: Hmm. Fair enough. My rubyisms aren't the best.

File: docs/resources/azure_security_center_policies.md.erb (hunk 20 above)
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azure_security_center_policies.md.erb]: Sounds good.

File: docs/resources/azurerm_virtual_machine.md.erb (hunk 27 above)
[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azurerm_virtual_machine.md.erb]: Ah, good point. My second comment came before I read yours.

[the remark this answers is not in the document]
Comment (jerryaldrichiii) [reply docs/resources/azurerm_virtual_machine.md.erb]: I'd like to avoid sort though. A count would work, but I'll leave it to you.

File: docs/resources/azurerm_virtual_machines.md.erb (hunk 5 above)
Comment (jerryaldrichiii) [reply docs/resources/azurerm_virtual_machines.md.erb]: Hmmm...not at the moment, you're right. Array of array's is tricky. Disregard my comment :smile:.

File: libraries/azure_network_security_group.rb (hunk 11 above)
Comment (jerryaldrichiii) [reply libraries/azure_network_security_group.rb]: Yip :smile:

File: libraries/azure_security_center_policy.rb (hunk 12 above)
Comment (jerryaldrichiii) [reply libraries/azure_security_center_policy.rb]: You da best.

File: inspec.lock [hunk 31]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (clintoncwolfe) [inline inspec.lock]: We should not ship the inspec lockfile with the resource pack.

File: inspec.yml [hunk 32]
```diff
@@ -1,7 +1,11 @@
...
-copyright: russell.seymour@turtlesystems.co.uk
-copyright_email: russell.seymour@turtlesystems.co.uk
-version: 0.2.0
\ No newline at end of file
+maintainer: Chef Software, Inc.
+copyright: Chef Software, Inc.
+copyright_email: support@chef.io
+license: Apache-2.0
+summary: This resource pack provides compliance resources for Azure.
+version: 1.0.0
```
Comment (clintoncwolfe) [inline inspec.yml]: :100: for the semver major bump :)

File: test/functional/verify/controls/azure_network_security_group.rb [hunk 33]
```diff
@@ -0,0 +1,24 @@
+resource_group = attribute('resource_group',            default: nil)
```
Comment (clintoncwolfe) [inline test/functional/verify/controls/azure_network_security_group.rb]: For consistency, we may eventually want to reclass these tests as "integration" tests (that's what the inspec project calls any test that does live-fire tests against a specific platform target).  inspec project uses the word 'functional' to refer specifically to UX testing.  Not a blocker issue.

File: bin/inspec [hunk 34]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
```
[the remark this answers is not in the document]
Comment (dmccown) [reply bin/inspec]: This was needed when we had continuous integration setup. If you have `chefdk` installed you can end up in a situation where the version of InSpec you bundle with will be different that the version of InSpec included in ChefDK. The binstub guarantees that you will always run with the version of InSpec you are bundled with.

File: docs/resources/azurerm_virtual_machine.md.erb [hunk 35]
```diff
@@ -0,0 +1,172 @@
+---
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azurerm_virtual_machine.md.erb]: At the moment it was just to avoid an implementation collision between the existing azure_virtual_machine resource and this implementation. I can update the docs to make that more clear, or we could rename all the new resources to the `azurerm` namespace if that would be more clear.

`azurerm` seems the preferred namespace in Terraform.

File: docs/resources/azure_generic_resource.md [hunk 36]
```diff
@@ -1,81 +0,0 @@
----
```
[the remark this answers is not in the document]
Comment (dmccown) [reply docs/resources/azure_generic_resource.md]: What's the risk if we have an implementation in `inspec-core` and the implementation in the resource pack? If they diverge which implementation will the resource pack use?

File: test/functional/verify/controls/azure_network_security_group.rb (hunk 33 above)
Comment (dmccown) [reply test/functional/verify/controls/azure_network_security_group.rb]: We can rename the folder that's not a big deal.

File: docs/resources/azurerm_virtual_machine.md.erb (hunk 35 above)
[the remark this answers is not in the document]
Comment (trevorghess) [reply docs/resources/azurerm_virtual_machine.md.erb]: azure vs azurerm is a common thing. It denotes the separation between Azure classic resource (Azure Service Manager or ASM) vs Azure Resource Manager (ARM)

**Comment by jerryaldrichiii** [comment]
All of my feedback is addressed thanks @dmccown!
